src/SoftwareLicense.php

```diff
@@ -928,6 +928,25 @@ public function getSpecificMassiveActions($checkitem = null)
         return $actions;
     }
 
+    public function getForbiddenSingleMassiveActions()
+    {
+        $forbidden = parent::getForbiddenSingleMassiveActions();
+
+        $number = Item_SoftwareLicense::countForLicense($this->getID());
+        $number += SoftwareLicense_User::countForLicense($this->getID());
+
+        //If we don't have update rights OR we've reached/exceeded license limit without overquota permission
+        if (
+            !static::canUpdate()
+            || ($this->getField('number') != -1 && $number >= $this->getField('number')
+            && !$this->getField('allow_overquota'))
+        ) {
+            $forbidden[] = 'Item_SoftwareLicense' . MassiveAction::CLASS_ACTION_SEPARATOR . 'add_item';
+        }
```

Comment:
You should encapsulate all of this into a `if (!$this->fields['allow_overquota'])` block, to prevent SQL queries executions if this is not required.

Also, please avoid usage of the `getField()` method when you are not is a display context. This method was made to display values, with a fallback to `N/A`. Tests based on it may provide unexpected results.